protocols/smtp/src/main/java/org/apache/james/protocols/smtp/AllButStartTlsDelimiterChannelHandler.java

```diff
@@ -27,19 +27,19 @@
 import org.jboss.netty.channel.Channel;
 import org.jboss.netty.channel.ChannelHandlerContext;
 import org.jboss.netty.channel.ChannelPipeline;
-import org.jboss.netty.handler.codec.frame.DelimiterBasedFrameDecoder;
+import org.jboss.netty.handler.codec.frame.LineBasedFrameDecoder;
 
 import com.google.common.base.CharMatcher;
 import com.google.common.base.Splitter;
 
 
-public class AllButStartTlsDelimiterChannelHandler extends DelimiterBasedFrameDecoder {
+public class AllButStartTlsDelimiterChannelHandler extends LineBasedFrameDecoder {
 
     private static final String STARTTLS = "starttls";
     private final ChannelPipeline pipeline;
 
-    public AllButStartTlsDelimiterChannelHandler(ChannelPipeline pipeline, int maxFrameLength, boolean stripDelimiter, ChannelBuffer[] delimiters) {
-        super(maxFrameLength, stripDelimiter, delimiters);
+    public AllButStartTlsDelimiterChannelHandler(ChannelPipeline pipeline, int maxFrameLength, boolean stripDelimiter) {
+        super(maxFrameLength, stripDelimiter, stripDelimiter);
```

Comment:
The second boolean parameter correspond to "fail fast" which was false before. We should preserve that behaviour.

I propose we introduce a constant named `FAIL_FAST` with value being `true`.

Thus parent constructor call becomes:

```
super(maxFrameLength, stripDelimiter, !FAIL_FAST);
```